File: searching/find_sum.py
def find_sum(lst, n):
    lst.sort()
    low = 0
    high = len(lst) - 1
    while low != high:
        if lst[low] + lst[high] < n:
            low += 1
        elif lst[low] + lst[high] > n:
            high -= 1
        else:
            return [lst[low], lst[high]]

    return None

File: searching/findSum.cpp
#include <vector>
#include <algorithm>

int* findSum(std::vector<int>& vec, int& n)
{
    std::sort(vec.begin(), vec.end());
    int low = 0;
    int high = vec.size() - 1;

    while (low != high)
    {
        if (vec[low] + vec[high] < n)
            low++;
        else if (vec[low] + vec[high] > n)
            high--;
        else
        {
            int arr[2] = { vec[low], vec[high] };
            return arr;
        }
    }
}